pandas/plotting/_core.py

```diff
@@ -422,8 +427,11 @@ def get_label(i):
 
         elif self.orientation == 'horizontal':
             if self._need_to_set_index:
-                yticklabels = [get_label(y) for y in ax.get_yticks()]
+                yticks = ax.get_yticks()
+                yticklabels = [get_label(y) for y in yticks]
                 ax.set_yticklabels(yticklabels)
+                ax.xaxis.set_major_locator(FixedLocator(yticks))
+                ax.xaxis.set_major_formatter(FixedFormatter(yticklabels))
```

Comment:
Why is the **x** axis given the **y**-tick(label)s here? Seems like an error to me. 
Meaning the "horizontal" direction is not sufficiently tested.

Also note that `ax.set_xticklabels(xticklabels)` and `ax.xaxis.set_major_formatter(FixedFormatter(xticklabels))` are redundant. One of those lines should be enough.

Reply:
This is definitly an error. 
And this part is definitly not tested at all. None of the tests in plotting even hit this lines. This is because the self._need_to_set_index is only set to True for plot that have a self.orientation "vertical", so there is no way to go inside this loop.
What is the best course of action? Deleting this part that could never get hits?

Reply:
If you have identified dead code you can submit a follow up PR to remove